monitor/sth_fetcher.go

```diff
@@ -75,8 +99,66 @@ type sthFetcher struct {
 
 	// How long to sleep between fetching the log's current STH
 	sthFetchInterval time.Duration
-	// How long to wait before giving up on an STH fetch
+	// How long to wait before giving up on an STH fetch or an STH consistency
+	// proof.
 	sthTimeout time.Duration
+
+	// prevSTH is the last STH that was fetched from the log
+	prevSTH *ct.SignedTreeHead
+
+	// sthLock is a Mutex for controlling updates to prevSTH
+	sthLock sync.Mutex
+
+	// verifier is used by verifySTHConsistency to prove consistency between two
+	// STHs
+	verifier sthFetcherVerifier
+}
+
+// newSTHFetcher returns an sthFetcher instance populated based on the provided
+// arguments
+func newSTHFetcher(
+	logger *log.Logger,
+	clk clock.Clock,
+	client monitorCTClient,
+	logURI string,
+	sthFetchInterval time.Duration,
+	sthTimeout time.Duration) *sthFetcher {
+	return &sthFetcher{
+		logger:           logger,
+		clk:              clk,
+		stats:            sthStats,
```

Comment:
Let's sort these so the stuff copied from the function args is all in one group, in the same order as the args, then the stuff that is constructed or copied from the global scope comes second. Makes it easier to see if all params are used.

Reply:
Sorted in ed2c6b2